src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -558,4 +626,41 @@ private void validateDatasetDacAccess(User user, Dataset dataset) {
       }
     }
   }
+
+  /**
+   * Determine if the user is a dataset/study creator, or if they are listed as a data custodian.
+   *
+   * @param user    User
+   * @param dataset Dataset
+   * @return User is a creator or custodian of the dataset.
+   */
+  protected boolean isCreatorOrCustodian(User user, Dataset dataset) {
```

Comment:
What would you think about breaking this into two separate methods? Would we ever need to individually check if a user is a creator vs a custodian?

Reply:
There are other places in the code where we look for custodians, but this is the only place where we care this specific scenario. I could move this to the `Dataset` object too, that would make it more useful if we did need it elsewhere in the future.